### PR TITLE
Add MSAVI index to backend

### LINKED ISSUE #30: Requesting same end date as current cache end date crashes server.
Requesting an end date that equals the cache's end date results in an empty range request to GEE, which returns an unhandled exception, and crashes our server.
- "equal dates" case is not handled for either start or end dates
- overlap case increases `processing_start_date` above `processing_end_date`

### PR DESCRIPTION
**Pull request by wherop**
- add endpoint to API for requesting MSAVI
- refactor index services, to consider which index was requested
- get MSAVI info from GEE
- add static cache of MSAVI data (same date range as NDVI)
- get MSAVI info from cache
- some clean-up and polish
- fix #30

### CODE REVIEW

File: backend/src/routes/sat_index_router.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    location: LocationName = Query(..., description="Name of the requested location"),
+    temporalResolution: TemporalResolution = Query(
+        ...,
+        description="Time interval that a single data point should represent e.g. one month",
+    ),
+    aggregation: AggregationMethod = Query(
+        ...,
+        description="Method of aggregating available data into a single datapoint to represent the selected time interval e.g. mean average",
+    ),
+):
```
Comment (tomADC443) [inline backend/src/routes/sat_index_router.py:95]: in the long run we might wanna put the schema for input validation into a seperate file which would probably lead to less code and more reusability :)

**Comment by tomADC443** [comment]
Very nice generalization of the existing architecture to work with all indexes 👍

**Comment by tomADC443** [comment]
i liked how you made the cache 'writeable' and thanks a lot for all the refactoring and clean up!

**Review by tomADC443 — APPROVED** [review]

File: backend/src/routes/sat_index_router.py (hunk 1 above)
Comment (wherop) [reply backend/src/routes/sat_index_router.py:95]: Agreed. I encountered many things like this that I want to change. I decided not to include them in the scope of this branch.

**Pull request merged**